mlfinlab/feature_importance/fingerpint.py

```diff
@@ -0,0 +1,259 @@
+"""
+Implementation of an algorithm described in Yimou Li, David Turkington, Alireza Yazdani
+'Beyond the Black Box: An Intuitive Approach to Investment Prediction with Machine Learning'
+(https://jfds.pm-research.com/content/early/2019/12/11/jfds.2019.1.023)
+"""
+
+from abc import ABC, abstractmethod
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.linear_model import LinearRegression
+
+
+# pylint: disable=invalid-name
+
+class AbstractModelFingerprint(ABC):
+    """
+    Model fingerprint constructor.
+    This is an abstract class, use RegressionModelFingerprint or ClassificationModelFingerprint.
+
+    :param model: (object) trained regression model
+    :param X: (pd.DataFrame) of features
+    :param num_values: (int) number of values used to estimate feature effect
+    """
+
+    def __init__(self, model: object, X: pd.DataFrame, num_values=50):
+        """
+        Model fingerprint constructor.
+        """
+
+        self.X = X
+        self.model = model
+
+        # Effects containers
+        self.linear_effect = None
+        self.non_linear_effect = None
+        self.pair_wise_effect = None
+
+        self.num_values = num_values
+
+        # Get possible feature values (Step 1)
```

Comment:
Lets move these computations to the fit method.

I think the following is more intuitive for the end user:

`model.fit() -> model.fit(model, X, num_values=100)`